src/app/dialog/attribute-function/attribute-function-dialog.component.ts

```diff
@@ -0,0 +1,132 @@
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {ChangeDetectionStrategy, Component, OnInit} from '@angular/core';
+import {Observable} from 'rxjs';
+import {Attribute, Collection} from '../../core/store/collections/collection';
+import {ActivatedRoute} from '@angular/router';
+import {DialogService} from '../dialog.service';
+import {select, Store} from '@ngrx/store';
+import {filter, map, mergeMap, tap} from 'rxjs/operators';
+import {selectAllCollections, selectCollectionById} from '../../core/store/collections/collections.state';
+import {CollectionsAction} from '../../core/store/collections/collections.action';
+import {BLOCKLY_VALUE_TOOLBOX} from '../../shared/blockly/blockly-editor/blockly-editor-toolbox';
+import {RuleVariable} from '../../collection/settings/tab/rules/rule-variable-type';
+import {MasterBlockType} from '../../shared/blockly/blockly-editor/blockly-editor.component';
+import {LinkType} from '../../core/store/link-types/link.type';
+import {selectLinkTypesByCollectionId} from '../../core/store/common/permissions.selectors';
+import {BlocklyDebugDisplay} from '../../shared/blockly/blockly-debugger/blockly-debugger.component';
+
+@Component({
+  selector: 'attribute-function-dialog',
+  templateUrl: './attribute-function-dialog.component.html',
+  styleUrls: ['./attribute-function-dialog.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AttributeFunctionDialogComponent implements OnInit {
+  public valueToolbox = BLOCKLY_VALUE_TOOLBOX;
+  public masterValueType = MasterBlockType.Value;
+  public variables$: Observable<RuleVariable[]>;
+
+  public collections$: Observable<Collection[]>;
+  public collection$: Observable<Collection>;
+  public attribute$: Observable<Attribute>;
+  public linkTypes$: Observable<LinkType[]>;
+
+  public displayDebug: BlocklyDebugDisplay;
+  public debugButtons: BlocklyDebugDisplay[] = [BlocklyDebugDisplay.DisplayJs, BlocklyDebugDisplay.DisplayError];
+
+  public js: string = '';
+  private xml: string = '';
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private dialogService: DialogService,
+    private store$: Store<{}>
+  ) {}
+
+  public ngOnInit() {
+    this.collections$ = this.store$.select(selectAllCollections);
```

Comment:
Do you really need all collections?

Reply:
Yes, for blockly to be able to display collection icons and colors. It can in theory meet anything...